packages/front-end/components/MetricAnalysis/MetricAnalysis.tsx

```diff
@@ -0,0 +1,784 @@
+  if (metricType === "ratio" && numeratorFormatter && denominatorFormatter) {
+    numeratorText = "Numerator: ";
+    denominatorText = <>{"Denominator: "}</>;
+    numeratorValue = <>{numeratorFormatter(result.numerator ?? 0)}</>;
+    denominatorValue = <>{denominatorFormatter(result.denominator ?? 0)}</>;
+  } else {
+    numeratorText = "Metric Total: ";
+    numeratorValue = (
+      <>
+        {metricType === "proportion"
+          ? formatNumber(result.units * result.mean)
+          : formatter(result.units * result.mean)}
+      </>
+    );
+    denominatorText = (
+      <>
+        {"Unique "}
+        <code>{userIdType}</code>
+        {": "}
+      </>
+    );
+    denominatorValue = <>{formatNumber(result.units)}</>;
+  }
+  return (
+    <div className="mb-4">
+      <div className="row mt-3">
+        <div className="col-auto">
+          <h4 className="mb-3 mt-1">{name}</h4>
+        </div>
+      </div>
+      <div className="d-flex flex-row align-items-end">
+        <div className="ml-0 appbox p-3 text-center row align-items-center">
+          <div className="col-auto">
+            <div className="border-bottom row">
+              <div className="mr-2">{numeratorText}</div>
+              <div style={{ flex: 1 }} />
+              <div>{numeratorValue}</div>
+            </div>
+            <div className="row">
+              <div className="mr-2">{denominatorText}</div>
+              <div style={{ flex: 1 }} />
+              <div>{denominatorValue}</div>
+            </div>
+          </div>
+          <div className="col-auto" style={{ fontSize: "2.5em" }}>
+            {"="}
+          </div>
+          <div className="col-auto">
+            <div style={{ fontSize: "2.5em" }}>{formatter(result.mean)}</div>
+            {metricType === "ratio" ? null : (
+              <>
+                {metricType === "proportion" ? "of" : "per"}{" "}
+                <code>{userIdType}</code>
+              </>
+            )}
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+}
+
+function getLookbackSelected(lookbackDays: number): string {
+  return [7, 14, 30].includes(lookbackDays) ? `${lookbackDays}` : `custom`;
+}
+
+function getDesiredSettings(
+  id: string,
+  values: MetricAnalysisFormFields,
+  endOfToday: Date
+): CreateMetricAnalysisProps {
+  const todayMinusLookback = new Date(endOfToday);
+  todayMinusLookback.setDate(
+    todayMinusLookback.getDate() - (values.lookbackDays as number)
+  );
+  todayMinusLookback.setHours(0, 0, 0, 0);
+
+  return {
+    id: id,
+    userIdType: values.userIdType,
+    lookbackDays: Number(values.lookbackDays),
+    startDate: todayMinusLookback.toISOString().substring(0, 16),
+    endDate: endOfToday.toISOString().substring(0, 16),
+    populationType: values.populationType,
+    populationId: values.populationId ?? null,
+  };
+}
+
+function settingsMatch(
+  settings: MetricAnalysisSettings,
+  desiredSettings: CreateMetricAnalysisProps
+) {
+  return (
```

Comment:
not sure if this is going to extend but this is a way to make it so that if you add settings it will be more robust
```
  Object.entries(settings).every(
    ([key, value]) => desiredSettings[key] === value
  )
```

Reply:
that will just return a true or false if they match or not

Reply:
Thanks, safer now. I excluded a few fields that I don't want to check